library/Zend/Validator/DateStep.php

```diff
@@ -169,7 +169,13 @@ protected function convertString($value, $addErrors = true)
             $date = new DateTime;
             $date->setISODate($matches[1], $matches[2]);
         } else {
-            $date = DateTime::createFromFormat($this->format, $value, $this->timezone);
+            $format = $this->format;
+
+            if(stripos($format, '!') === false) {
```

Comment:
Also, no need for the case insensitive variant, as you're working with a symbol, not a letter.